pkg/helm/export.go

```diff
@@ -26,7 +27,20 @@ func ExportHelmChart(rootFs, fs billy.Filesystem, helmChartPath string, chartVer
 	if err := chart.Validate(); err != nil {
 		return fmt.Errorf("Failed while trying to validate Helm chart: %s", err)
 	}
-	chartVersion = chart.Metadata.Version + chartVersion
+	chartVersionSemver, err := semver.Make(chart.Metadata.Version)
+	if err != nil {
+		return fmt.Errorf("Cannot parse original chart version %s as valid semver", chart.Metadata.Version)
+	}
+	// Add packageVersion as string, preventing errors due to leading 0s
+	if packageVersion >= 999 {
+		return fmt.Errorf("Maximum number of packageVersions is 999, found %d", packageVersion)
+	}
+	chartVersionSemver.Patch = 1000*chartVersionSemver.Patch + uint64(packageVersion)
```

Comment:
I chose 1000 since that gives us the ability to handle up to 999 packageVersion changes and automatically ensures that any of the newly generated charts always have a bigger semver than the old charts unless they have a patchVersion of 0.

Reply:
We might nee to be careful about this. Does it mean the future chart will have something like x.y.z0000? This is probably not going to work well with old charts. Like for example if this is going to apply to old chart let's say 0.3.4, then it will become 0.3.4000 which will become the latest version if we don't update other version because 0.3.4000 is greater than 0.3.500 or 0.3.600. But as long as we are only updating the latest chart we should be good, but sounds like a big version change in our chart today.

Reply:
> Does it mean the future chart will have something like x.y.z000?

yes that was my expectation. We could revert this back to a packageVersion with just 2 digits.

re:
> Like for example if this is going to apply to old chart let's say 0.3.4, then it will become 0.3.4000 which will become the latest version if we don't update other version because 0.3.4000 is greater than 0.3.500 or 0.3.600. 

You could just patch the chart version from 0.3.4 => 0.3.0 and increment the packageVersion to 400. Due to the way that the new logic works for versioning (only applies on export), that would result in the right patchVersion you want of 0.3.400.

Reply:
@cbron any thoughts? I'm fine to revert back to 2 digit packageVersions.

Reply:
Yea lets do 2 digits, if for nothing else because everyone is familiar with it and there's less edge cases.